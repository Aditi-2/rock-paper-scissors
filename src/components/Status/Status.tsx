import React from "react";
import { GameStatus, GameType } from "../../common/types/game";

interface StatusProps {
  gameType: GameType,
  gameStatus: GameStatus;
}

export const Status: React.FC<StatusProps> = ({gameType, gameStatus}) => {
  const player1 = gameType === "PLAYER_VS_COMPUTER" ? "You" : "Computer one";
  const player2 =
    gameType === "PLAYER_VS_COMPUTER" ? "Computer" : "Computer two";

  return (
    <>
      <p>
        {gameStatus === "playerOne" && `${player1}  wins!`}
        {gameStatus === "playerTwo" && `${player2}  wins!`}
        {gameStatus === "DRAW" && "Draw!"}
      </p>
    </>
  );
};
