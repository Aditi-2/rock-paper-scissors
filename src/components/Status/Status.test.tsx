import React from 'react';
import {create} from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Status } from './Status';

const getComponent = () => <Status gameType="PLAYER_VS_COMPUTER" gameStatus="DRAW" />

test('Status Component snapshot', () => {
    const component = create(getComponent()).toJSON();
    expect(component).toMatchSnapshot();
});

test('renders Status Component', () => {
  render(getComponent());
  const statusText = screen.getByText(/Draw!/i);
  expect(statusText).toBeInTheDocument();
});

