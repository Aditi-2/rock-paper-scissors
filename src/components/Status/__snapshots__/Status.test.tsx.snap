// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Component snapshot 1`] = `
<p>
  Draw!
</p>
`;
